build-support/bin/release.sh

```diff
@@ -11,17 +11,14 @@ source "${ROOT}/build-support/common.sh"
 
 # Note we allow the user to predefine this value so that they may point to a specific interpreter.
 export PY="${PY:-python3.6}"
-interpreter_constraint="CPython==3.6.*"
 if ! command -v "${PY}" >/dev/null; then
   die "Python interpreter ${PY} not discoverable on your PATH."
 fi
 py_major_minor=$(${PY} -c 'import sys; print(".".join(map(str, sys.version_info[0:2])))')
-if [[ "${py_major_minor}" != "3.6" ]]; then
-  die "Invalid interpreter. The release script requires Python 3.6 (you are using ${py_major_minor})."
+if [[ "${py_major_minor}" != "3.6" || "${py_major_minor}" != "3.7" ]]; then
+  die "Invalid interpreter. The release script requires Python 3.6 or 3.7 (you are using ${py_major_minor})."
 fi
 
-export PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS="['${interpreter_constraint}']"
```

Comment:
I think that this can be removed now - I we don't care if subprocesses are ran with Py36 or Py37. The main reason I think we had this was with Python 2 to ensure that when `-3` was specified we didn't unintentionally run Pants with Py2.

Still, a little unclear if this is 100% safe.